server/src/main/java/org/apache/cassandra/sidecar/db/RestoreRange.java

```diff
@@ -486,6 +487,30 @@ private <T> T readSliceProperty(Function<RestoreSlice, T> func)
         return func.apply(source);
     }
 
+    private void updateRangeStatusForInstance(InstanceMetadata instance, RestoreRangeStatus status)
+    {
+        // skip if the belong job is not managed by sidecar
+        if (!job().isManagedBySidecar())
+        {
+            return;
+        }
+
+        statusByReplica.put(storageAddressWithPort(instance), status);
+    }
+
+    private String storageAddressWithPort(InstanceMetadata instance)
+    {
+        try
+        {
+            InetSocketAddress storageAddress = instance.getFromDelegate(CassandraAdapterDelegate::localStorageBroadcastAddress);
+            return storageAddress.getAddress().getHostAddress() + ':' + storageAddress.getPort();
```

Comment:
will this work? I think internally it does something somewhat similar, with the added benefit that will check if addr != null
```suggestion
            return return storageAddress.toString();
```